### PR TITLE
[WIP] Replace potrace with imagetracerjs

### LINKED ISSUE #17584: Incompatible Licenses Apache2.0 & GPLv2 (gatsby-plugin-sharp)
## Summary

Hello! 
Because reasons I had to make sure all licenses in a certain gatsby project we have play well together. Unfortunately there is an issue.

As explained here https://www.apache.org/licenses/GPL-compatibility.html 

> the FSF has never considered the Apache License to be compatible with GPL version 2

There are several apache2.0 licensed packages in a typical gatsby project, eg. starter has 42
and one library which is GPLv2 licensed, so this is where I see a problem.

This specific library is https://github.com/Iwasawafag/node-potrace
I opened an [issue](https://github.com/Iwasawafag/node-potrace/issues/7) and a [PR](https://github.com/Iwasawafag/node-potrace/pull/9) regarding this topic, no feedback so far.

node-potrace is a dependency of [gatsby-plugin-sharp](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-plugin-sharp/package.json). [Sharp](https://www.npmjs.com/package/sharp) itself is apache2.0 licensed so I think the incompatibility is already on that level. Gatsby-plugin-sharp is licensed as MIT, however I miss the knowledge to evaluate if this is ok.

## Relevant information

FWIW I used https://github.com/davglass/license-checker and run it on the gatsbyjs/gatsby-starter-default and printed the licenses.

Here is the summary: https://github.com/einSelbst/gatsby-starter-default/blob/master/licenses-summary.txt

Here are the details: https://github.com/einSelbst/gatsby-starter-default/blob/master/licenses-details.json

## Question

I would like to know if gatsby thinks this is a relevant issue. I would think it's also in gatsby's own interest to make sure the code which is distributed doesn't infringe licenses.

### PR DESCRIPTION
**Pull request by davidbailey00**
## Description

Replace `potrace` (GPLv2 licensed) with `imagetracerjs` (public domain) for similar functionality

## Related Issues

Fixes #17584

### CODE REVIEW

**Comment by davidbailey00** [comment]
n.b. I'm currently building/testing against a local fork of `imagetracerjs` to add support for Node.js, so tests will fail on CI at the moment

File: packages/gatsby-plugin-sharp/src/trace-svg.js [hunk 1]
```diff
@@ -61,6 +62,13 @@ exports.notMemoizedPrepareTraceSVGInputFile = async ({
       if (err) {
         return reject(err)
       }
+      console.log(`created   ${tmpFilePath}`)
+      try {
+        accessSync(tmpFilePath)
+        console.log(`success`)
+      } catch (e) {
+        console.log(`fail`)
```
Comment (muescha) [inline packages/gatsby-plugin-sharp/src/trace-svg.js]: I guess there is not so much console log allowed

**Comment by muescha** [comment]
You use node canvas - I read here about a problem with canvas in ssr: #17661

**Comment by davidbailey00** [comment]
Hi @muescha! In this case I'm not using canvas for SSR, just for the purpose of generating SVGs on the server, so there is nothing to worry about.

Re: your comment on the `console.log`s, I used those to test what appears to be an issue with Sharp where it's not outputting files correctly - still trying to figure out what's happening.

**Comment by wardpeet** [comment]
The problem I see is the same as blurhash. Node-canvas needs GYP to built on post-install. So this might make gatsby-plugin-sharp less reliable.

**Comment by chan360** [comment]
Any updates on the current status of this PR?

**Comment by chan360** [comment]
@LekoArts I'm wondering if you have more information on this regarding that this is a "Maybe Fix" for https://github.com/gatsbyjs/gatsby/issues/17584

**Comment by LekoArts** [comment]
Since potracerjs doesn't have feature parity, closing this for now as there wouldn't be any progress on this anyways.

**Pull request closed without merging**